File: src/main/java/client/messages/quest/ResignQuestRecordMessage.java
package client.messages.quest;

import client.player.quest.QuestState;
import util.packet.PacketWriter;

public class ResignQuestRecordMessage extends AbstractQuestRecordMessage {

    private final boolean completed;

    public ResignQuestRecordMessage(short questId, boolean completed) {
        super(questId);
        this.completed = completed;
    }

    @Override
    public QuestState getState() {
        return QuestState.NONE;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        super.encodeData(pw);
        pw.writeBool(completed);
    }
}

File: src/main/java/client/messages/quest/ResignQuestRecordMessage.kt
package client.messages.quest

import client.player.quest.QuestState
import util.packet.PacketWriter

class ResignQuestRecordMessage(questId: Short, private val completed: Boolean) : AbstractQuestRecordMessage(questId) {

    override val state: QuestState get() = QuestState.NONE

    override fun encodeData(pw: PacketWriter) {
        super.encodeData(pw)
        pw.writeBool(completed)
    }
}